pysc2/bin/replay_actions.py

```diff
@@ -248,53 +178,37 @@ def process_replay(self, controller, replay_data, map_data, player_id):
 
     feat = features.Features(controller.game_info())
 
-    self.stats.replay_stats.replays += 1
+    self.stats.parser.replays += 1
     self._update_stage("step")
     controller.step()
+    data = []
     while True:
-      self.stats.replay_stats.steps += 1
+      self.stats.parser.steps += 1
       self._update_stage("observe")
       obs = controller.observe()
 
-      for action in obs.actions:
-        act_fl = action.action_feature_layer
-        if act_fl.HasField("unit_command"):
-          self.stats.replay_stats.made_abilities[
-              act_fl.unit_command.ability_id] += 1
-        if act_fl.HasField("camera_move"):
-          self.stats.replay_stats.camera_move += 1
-        if act_fl.HasField("unit_selection_point"):
-          self.stats.replay_stats.select_pt += 1
-        if act_fl.HasField("unit_selection_rect"):
-          self.stats.replay_stats.select_rect += 1
-        if action.action_ui.HasField("control_group"):
-          self.stats.replay_stats.control_group += 1
-
-        try:
-          func = feat.reverse_action(action).function
-        except ValueError:
-          func = -1
-        self.stats.replay_stats.made_actions[func] += 1
-
-      for valid in obs.observation.abilities:
-        self.stats.replay_stats.valid_abilities[valid.ability_id] += 1
-
-      for u in obs.observation.raw_data.units:
-        self.stats.replay_stats.unit_ids[u.unit_type] += 1
-
-      for ability_id in feat.available_actions(obs.observation):
-        self.stats.replay_stats.valid_actions[ability_id] += 1
-
-      if obs.player_result:
+      #if parser.parse_step returns, whatever is returned is appended
+      #to a data list, and this data list is saved to a json file
+      #in the data_dir directory with filename = replay_name_player_id.json
+      parsed_data = self.stats.parser.parse_step(obs,feat,info)
+      if parsed_data:
+        data.append(parsed_data)
+
+      if obs.player_result:        
+        #save scraped replay data to file at end of replay if parser returns        
+        if data:
+          data_file = FLAGS.data_dir + replay_name + "_" + str(player_id) + '.json'
+          with open(data_file,'w') as outfile:
+            json.dump(data,outfile)
```

Comment:
Added to disk file saving at end of replay if custom parser parse_step returns data